stubs/keyboard/keyboard/__init__.pyi

```diff
@@ -0,0 +1,169 @@
+        | tuple[Literal["pending"], Literal["up"], Literal["hotkey"]]
+        | tuple[Literal["pending"], Literal["down"], Literal["hotkey"]]
+        | tuple[Literal["suppressed"], Literal["up"], Literal["hotkey"]]
+        | tuple[Literal["suppressed"], Literal["down"], Literal["hotkey"]]
+        | tuple[Literal["allowed"], Literal["up"], Literal["hotkey"]]
+        | tuple[Literal["allowed"], Literal["down"], Literal["hotkey"]]
+        | tuple[Literal["free"], Literal["up"], Literal["other"]]
+        | tuple[Literal["free"], Literal["down"], Literal["other"]]
+        | tuple[Literal["pending"], Literal["up"], Literal["other"]]
+        | tuple[Literal["pending"], Literal["down"], Literal["other"]]
+        | tuple[Literal["suppressed"], Literal["up"], Literal["other"]]
+        | tuple[Literal["suppressed"], Literal["down"], Literal["other"]]
+        | tuple[Literal["allowed"], Literal["up"], Literal["other"]]
+        | tuple[Literal["allowed"], Literal["down"], Literal["other"]],
+        tuple[Literal[False], Literal[True], Literal["free"]]
+        | tuple[Literal[False], Literal[False], Literal["pending"]]
+        | tuple[Literal[True], Literal[True], Literal["free"]]
+        | tuple[Literal[False], Literal[True], Literal["allowed"]]
+        | tuple[Literal[False], Literal[False], Literal["free"]]
+        | tuple[Literal[False], Literal[False], Literal["suppressed"]]
+        | tuple[Literal[False], None, Literal["free"]]
+        | tuple[Literal[False], None, Literal["suppressed"]]
+        | tuple[Literal[False], None, Literal["allowed"]]
+        | tuple[Literal[True], Literal[True], Literal["allowed"]]
+        | tuple[Literal[False], Literal[False], Literal["allowed"]],
+    ]
+    active_modifiers: set[int]
+    blocking_hooks: list[_Callback]
+    blocking_keys: defaultdict[int, list[_Callback]]
+    nonblocking_keys: defaultdict[int, list[_Callback]]
+    blocking_hotkeys: defaultdict[tuple[int, ...], list[_Callback]]
+    nonblocking_hotkeys: defaultdict[tuple[int, ...], list[_Callback]]
+    filtered_modifiers: Counter[int]
+    is_replaying: bool
+    modifier_states: dict[_Key, str]
+    def init(self) -> None: ...
+    def pre_process_event(self, event): ...
+    def direct_callback(self, event): ...
+    def listen(self) -> None: ...
+
+def key_to_scan_codes(key: _ParseableHotkey, error_if_missing: bool = ...) -> tuple[int, ...]: ...
+def parse_hotkey(hotkey: _ParseableHotkey) -> tuple[tuple[tuple[int, ...], ...], ...]: ...
+def send(hotkey: _ParseableHotkey, do_press: bool = ..., do_release: bool = ...) -> None: ...
+
+press_and_release = send
+
+def press(hotkey: _ParseableHotkey) -> None: ...
+def release(hotkey: _ParseableHotkey) -> None: ...
+
+# is_pressed cannot check multi-step hotkeys, so not using _ParseableHotkey
+
+def is_pressed(hotkey: _Key | _ScanCodeList) -> bool: ...
+def call_later(fn: Callable[..., None], args: _P = ..., delay: float = ...) -> None: ...
```

Comment:
This should be expressible with PEP 646's TypeVarTuple, though we can't use that yet because mypy doesn't support it.

Reply:
Good to know. I hadn't looked into 3.11's variadic generics yet. That's cool. So this would be dependant on https://github.com/python/mypy/issues/12280